apis/clusterresources/v1beta1/awsendpointserviceprincipal_webhook.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"fmt"
+	"regexp"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var awsendpointserviceprincipallog = logf.Log.WithName("awsendpointserviceprincipal-resource")
+
+func (r *AWSEndpointServicePrincipal) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-clusterresources-instaclustr-com-v1beta1-awsendpointserviceprincipal,mutating=false,failurePolicy=fail,sideEffects=None,groups=clusterresources.instaclustr.com,resources=awsendpointserviceprincipals,verbs=create;update,versions=v1beta1,name=vawsendpointserviceprincipal.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &AWSEndpointServicePrincipal{}
+
+var principalArnPattern, _ = regexp.Compile(`^arn:aws:iam::[0-9]{12}:(root$|user\/[\w+=,.@-]+|role\/[\w+=,.@-]+)$`)
```

Comment:
We can move this variable to validation.go file located in models package, wdyt?

Reply:
I don't think that this is a good idea. Every regexp pattern is individual and is used only for one specific validation, hence I declared it here